src/commands/status/masternode.js

```diff
@@ -19,23 +20,57 @@ class MasternodeStatusCommand extends BaseCommand {
     const rows = [];
 
     // Version
-    const versionOutput = await dockerCompose.execCommand(
+    const dashdVersion = (await dockerCompose.execCommand(
       config.toEnvs(),
       'core',
       'dashd --version',
-    );
+    )).out.split('\n')[0];
 
-    rows.push(['Version', versionOutput.out.split('\n')[0]]);
+    // Sync status
+    const mnsyncStatus = JSON.parse((await dockerCompose.execCommand(
```

Comment:
We can use RPC instead of calling through docker which is much slower. You can find examples in the project code.

Reply:
Like [this](https://github.com/dashevo/mn-bootstrap/pull/124/commits/761505be33b6f042ee8ee7698ae5e553b1012572)? Is something similar available for the Tendermint calls?

Reply:
Tendermint has RPC too, but in this case, we need to publish port such as for Core. We really don't want to do that. I didn't take it into account. Probably, it's better to use docker exec for now.

Reply:
OK, I have changed it back. Doesn't the `rpcallowip` we define restrict which hosts could access RPC? Maybe if we just define it more restrictively by removing (at least) `rpcallowip=192.168.0.0/16` it would be safer. Could we also randomly generate a different RPC password on startup each time?

Another solution I looked into to make the delay less noticeable was using [table streaming](https://github.com/gajus/table#streaming) but this output has a bug that doesn't support single line mode, so I don't like the appearance much.

Would it be possible to use `Promise.all()` to run the requests asynchronously? This would also provide a speedup by running all the `docker exec` commands simultaneously right?

Reply:
`rpcallowip=192.168.0.0/16` this needed for Docker, we can't remove it, but in general, I agree with you. Also, I agree that we need to generate Core RPC user and password and store it in an MN config.

Reply:
With `Promise.all` it should be better.

Reply:
Can configure Tendermint RPC for local requests as well? https://docs.tendermint.com/master/rpc/